crates/matrix-sdk/src/widget/mod.rs

```diff
@@ -50,6 +41,84 @@ pub struct Comm {
     pub to: Sender<String>,
 }
 
+/// Information about a widget.
+#[derive(Debug)]
+pub struct WidgetSettings {
+    /// Widget's unique identifier.
+    pub id: String,
+    /// Whether or not the widget should be initialized on load message
+    /// (`ContentLoad` message), or upon creation/attaching of the widget to
+    /// the SDK's state machine that drives the API.
+    pub init_on_load: bool,
+    /// This contains the url from the widget state event
+    /// In this url placeholders can be used to pass information from the client to the widget
+    /// Possible values are: `$widgetId`, `$parentUrl`, `$userId`, `$lang`, `$fontScale`, `$analyticsID`
+    /// 
+    /// # Examples
+    /// 
+    /// e.g `http://widget.domain?username=$userId`
+    /// will become: `http://widget.domain?username=@user_matrix_id:server.domain`
+    raw_url: Url,
```

Comment:
Shouldn't it be something like [`Uri`](https://docs.rs/http/latest/http/uri/struct.Uri.html)?

Reply:
what makes it a Uri? I think all the parts it can contain are conformant to a URL? Schme, Authority, Path, Query string, Fragment Identifier.